### PR TITLE
rpc: disable BigTable with header

### PR DESCRIPTION
**Pull request by fanatid**
#### Problem

By default, all RPC queries can reach BigTable if they need more data. In some cases, it can be good to control should we reach BigTable or not. For example, we can use cache layers after `solana-validator`, like `old-faithful` / S3-like storage / local disk. In https://triton.one/ we do this with `x-bigtable` header (our docs: https://docs.triton.one/chains/solana/old-faithful-historical-archive-1). Probably it would be good to add it to upstream so all providers can support this feature.

#### Summary of Changes

Add extra clone method for `x-bigtable` header.

moved from https://github.com/solana-labs/solana/pull/34975

### CODE REVIEW

File: rpc/src/rpc_service.rs [hunk 1]
```diff
@@ -525,7 +525,14 @@ impl JsonRpcService {
                 );
                 let server = ServerBuilder::with_meta_extractor(
                     io,
-                    move |_req: &hyper::Request<hyper::Body>| request_processor.clone(),
+                    move |req: &hyper::Request<hyper::Body>| {
+                        let xbigtable = req.headers().get("x-bigtable");
+                        if xbigtable.map(|v| v == "disabled").unwrap_or(false) {
```
Comment (CriesofCarrots) [inline rpc/src/rpc_service.rs]:
```suggestion
                        if xbigtable.is_some_and(|v| v == "disabled") {
```

Comment (fanatid) [reply rpc/src/rpc_service.rs]: nice, did not know about `is_some_and`! `matches!` also would be a good alternative to `map+unwrap_or`

**Review by CriesofCarrots — APPROVED** [review]
Lgtm!

The CI failure (`clippy-nightly (macos-latest-large)`) is not related, so I may merge on red, depending on how quickly that's resolved.

**Comment by fanatid** [comment]
Thank you! Would be nice to have backport to 1.18 :slightly_smiling_face:

**Pull request merged**